### PR TITLE
Automatic Scoping of HTTP requests/responses for dataflow components.

### PR DESCRIPTION
**Pull request by TylorS**
Add tests related to auto-scoping
Implement isolateSink && isolateSource to adhere to @cycle/isolate API
Change code style to reflect that of the rest of the Motorcycle libraries.

### CODE REVIEW

**Comment by TylorS** [comment]
Anyone see any reason why this shouldn't be merged?

**Comment by Frikki** [comment]
Here https://github.com/motorcyclejs/motorcycle-http/pull/1/files#diff-1fdf421c05c1140f6d71444ea2b27638R37

There are lot’s of if statements. This could be handled much cleaner with action objects.

Also here https://github.com/motorcyclejs/motorcycle-http/pull/1/files#diff-1fdf421c05c1140f6d71444ea2b27638R82

**Comment by TylorS** [comment]
I agree that could be much cleaner, I simply copied that part over from cycle-http-driver.

**Comment by TylorS** [comment]
I'll work on that now

**Comment by Frikki** [comment]
https://github.com/motorcyclejs/motorcycle-http/pull/1/files#diff-1fdf421c05c1140f6d71444ea2b27638R122

Don’t you have a function made to check this?

File: src/index.js [hunk 1]
```diff
@@ -2,123 +2,156 @@ import most from 'most'
...
+      if (headers.hasOwnProperty(key)) {
+        request = request.set(key, headers[key])
       }
     }
-  }
-  if (attach !== null) {
-    for (let i = attach.length - 1; i >= 0; i--) {
-      const a = attach[i]
-      request = request.attach(a.name, a.path, a.filename)
+    if (is.notNull(field)) {
```
Comment (Frikki) [inline src/index.js]: Is this check needed? If `field` is `null`, the loop won’t run anyway.

Comment (TylorS) [inline src/index.js]: Good point

Comment (TylorS) [inline src/index.js]: Done!

**Comment by Frikki** [comment]
When ready for review/merge, please assign.

**Comment by Frikki** [comment]
Merged into _develop_ branch.

**Comment by Frikki** [comment]
Can we delete _auto-scope_ branch?

**Comment by TylorS** [comment]
Yes

**Pull request closed without merging**